sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -329,23 +386,27 @@ def __init__(self, message=None, response=None, **kwargs):
         super(HttpResponseError, self).__init__(message=message, **kwargs)
 
     @staticmethod
-    def _parse_odata_body(error_format: Type[ODataV4Format], response: "_HttpResponseBase") -> Optional[ODataV4Format]:
-        try:
-            odata_json = json.loads(response.text())
-            return error_format(odata_json)
-        except Exception:  # pylint: disable=broad-except
-            # If the body is not JSON valid, just stop now
-            pass
+    def _parse_odata_body(
+        error_format: Type[ODataV4Format], response: Optional[_HttpResponseCommonAPI]
+    ) -> Optional[ODataV4Format]:
+        if response:
+            try:
+                odata_json = json.loads(response.text())
+                return error_format(odata_json)
+            except Exception:  # pylint: disable=broad-except
+                # If the body is not JSON valid, just stop now
+                pass
         return None
 
-    def __str__(self):
+    def __str__(self) -> str:
         retval = super(HttpResponseError, self).__str__()
-        try:
-            body = self.response.text()
-            if body and not self.error:
-                return "{}\nContent: {}".format(retval, body)[:2048]
-        except Exception:  # pylint: disable=broad-except
-            pass
+        if self.response:
```

Comment:
Was this to make mypy happy? Seems superfluous.

Reply:
Yes, response is optional and mypy wasn't catching that the AttributeError will be caught by the wide exception. If you prefer a type ignore I can do that?

Reply:
https://github.com/python/mypy/issues/14743#issuecomment-1664725053